internal/handlers/pg/msg_insert.go

```diff
@@ -95,6 +95,33 @@ func (h *Handler) MsgInsert(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg,
 	return &reply, nil
 }
 
+func insertMany(ctx context.Context, dbPool *pgdb.Pool, qp *pgdb.QueryParams, docs *types.Array, ordered bool) (int32, *commonerrors.WriteErrors) { //nolint:lll // argument list is too long
+	var inserted int32
+	var insErrors commonerrors.WriteErrors
+
+	dbPool.InTransaction(ctx, func(tx pgx.Tx) error {
+        var err error
+        for i := 0; i < docs.Len(); i++ {
+            doc := must.NotFail(docs.Get(i))
+            err = insertDocument(ctx, dbPool, qp, doc)
+            var we *commonerrors.WriteErrors
+            switch {
+            case err == nil:
+                inserted++
```

Comment:
In principle, we don't need to calculate it: as it's a transition, we insert all or nothing.

Reply:
Yes that is true, i will discard it in the single transaction run